# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1689893447 -7200
# Parent  ee76879b5e1bf51b8537a6d61bfa419e41a2cbb6
Feature: Re-implementation of the two-line view: Make correspondents column pixel perfect.

diff --git a/mail/themes/shared/mail/threadPane.css b/mail/themes/shared/mail/threadPane.css
--- a/mail/themes/shared/mail/threadPane.css
+++ b/mail/themes/shared/mail/threadPane.css
@@ -371,16 +371,19 @@ tr[is="thread-row"][multiline="true"] td
 tr[is="thread-row"][multiline="true"] td.subjectcol-column {
   overflow: visible !important;
 }
 
 /* Indent "alignment column" as well. */
 tr[is="thread-row"][multiline="true"] td[alignmentcolumn="true"] {
   padding-left: calc(8px + 16px * var(--thread-level));
 }
+tr[is="thread-row"][multiline="true"] td.correspondentcol-column[alignmentcolumn="true"] {
+  padding-left: calc(6px + 16px * var(--thread-level));
+}
 
 /* Remove centering of the td */
 tr[is="thread-row"][multiline="true"] td {
   vertical-align: top;
 }
 :root[uidensity="compact"] tr[is="thread-row"][multiline="true"] td {
   padding-top: 2px;
 }
@@ -631,30 +634,37 @@ tr[data-properties~="junk"] :is(.tree-vi
   color: inherit;
 }
 
 /* Delete column variations */
 /* TODO: Handle delete button color variations for tagged messages */
 
 /* Correspondent column variations */
 
-#correspondentColButton,
 .correspondentcol-column {
   text-indent: 18px;
 }
+#correspondentColButton {
+  text-indent: calc(18px + var(--tree-header-cell-padding));
+}
 
 .correspondentcol-column {
   background-repeat: no-repeat;
   background-position-x: 1px;
   background-position-y: center;
   -moz-context-properties: stroke, fill-opacity;
   stroke: currentColor;
   fill-opacity: 0.3;
 }
 
+tr[is="thread-row"][multiline="true"] .correspondentcol-column {
+  background-position-x: calc(6px + 16px * var(--thread-level));
+  background-position-y: top 1px;
+}
+
 .correspondentcol-column:dir(rtl) {
   background-position-x: right 1px;
 }
 
 tr[data-properties~="outgoing"] .correspondentcol-column {
   background-image: var(--icon-nav-forward);
 }
 
